frontend/src/lib/components/FlaggedFeature.tsx

```diff
@@ -2,14 +2,14 @@ import { useValues } from 'kea'
 
 import { FeatureFlagKey } from 'lib/constants'
 import { useFeatureFlag } from 'lib/hooks/useFeatureFlag'
-import { featureFlagLogic } from 'lib/logic/featureFlagLogic'
+import { featureFlagLogic, getFeatureFlagPayload } from 'lib/logic/featureFlagLogic'
 
 export type PostHogFeatureProps = {
     flag: FeatureFlagKey
     /** What specific state or variant of feature flag needs to be active. */
     match?: string | boolean
     /** Rendered when the flag state/variant matches. */
-    children: React.ReactNode | ((payload: any) => React.ReactNode)
+    children: React.ReactNode | ((flagValue: string | boolean, payload: any) => React.ReactNode)
```

Comment:
Oh, this is a completely different type, it's surprising this didn't raise any TS errors 🤔